app/models/manageiq/providers/hawkular/middleware_manager/refresh_parser.rb

```diff
@@ -26,12 +28,60 @@ def get_middleware_servers
           @ems.eaps(feed).map do |eap|
             @eaps << eap
             server = parse_middleware_server(eap)
+
+            machine_id = @ems.machine_id(eap.feed)
+            host_instance = find_host_by_bios_uuid(machine_id)
+            if host_instance.nil?
+              host_instance = find_host_by_bios_uuid(alternate_machine_id(machine_id))
+            end
```

Comment:
you can collapse that to a one liner
`host_instance = find_host_by_bios_uuid(machine_id) || find_host_by_bios_uuid(alternate_machine_id(machine_id))`